### PR TITLE
CSU-1004: handle http timeouts

### PR DESCRIPTION
**Pull request by ValyaB**

### CODE REVIEW

**Review by stgleb — APPROVED** [review]
LGTM

File: main.go [hunk 1]
```diff
@@ -65,13 +68,20 @@ func main() {
 		},
 	}
 
+	log := logrus.WithFields(logrus.Fields{})
+
+	// Create a new client to communicate with CAST AI API.
+	deltaHTTPClient, err := castai.NewDefaultClient(cfg.API.URL, cfg.API.Key, logger.Level, binVersion, defaultPollTimeout)
```
Comment (andrejatcastai) [inline main.go:74]: I suppose this is copy-paste typo

File: main.go [hunk 2]
```diff
@@ -65,13 +68,20 @@ func main() {
...
+	if err != nil {
+		log.Fatalf("new http client failed: %v", err)
+	}
+
 	client := castai.NewClient(
 		logger,
-		castai.NewDefaultClient(cfg.API.URL, cfg.API.Key, logger.Level, binVersion),
+		deltaHTTPClient,
```
Comment (andrejatcastai) [inline main.go:81]: I can see it is not your change, but why here it is `logger`, whereas above it `log := logrus.WithFields(logrus.Fields{})`?

File: castai/client.go [hunk 3]
```diff
@@ -33,17 +36,54 @@ func NewClient(log *logrus.Logger, rest *resty.Client, clusterID string) Client
...
+			Timeout: 15 * time.Second,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   5 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
+
+	t2, err := http2.ConfigureTransports(t1)
```
Comment (andrejatcastai) [inline castai/client.go:73]: We are not using http2 in controller. Could you elaborate why http2?

Comment (ValyaB) [reply castai/client.go:73]: we are, it is default from resty library

File: main.go (hunk 1 above)
Comment (ValyaB) [reply main.go:74]: Where? Can you explain, please

**Comment by ValyaB** [comment]
Without configured tls, http2 is not used,  and 5m timeout is by server design

File: main.go (hunk 1 above)
Comment (apasyniuk) [reply main.go:74]: `deltaHTTPClient` we don't have deltas in the controller

File: main.go (hunk 2 above)
Comment (andrejatcastai) [reply main.go:81]: There are two logger objects created (logger and log). Could you explain why it is needed?

File: castai/client.go (hunk 3 above)
Comment (andrejatcastai) [reply castai/client.go:73]: I could not find Resty is calling `http2.ConfigureTransports`. Moreover to my understanding Resty relies on `http` (even though it sets `ForceAttemptHTTP2` to true) package: https://github.com/go-resty/resty/blob/v2/go.mod#L6

File: main.go (hunk 2 above)
Comment (ValyaB) [reply main.go]: It probably was added to avoid additional fields and log levels from env when printing fatal in the main.

**Pull request merged**